cmd/build.go

```diff
@@ -157,6 +184,87 @@ func build(projectName string, targetOS string, vmArguments []string) {
 	} else {
 		engineCachePath = enginecache.ValidateOrUpdateEngine(targetOS)
 	}
+	dockerFilePath, err := filepath.Abs(filepath.Join(tmpDir, "Dockerfile"))
+	if err != nil {
+		fmt.Printf("hover: Failed to resolve absolute path for Dockerfile %s: %v\n", dockerFilePath, err)
+		os.Exit(1)
+	}
+
+	dockerFile, err := os.Create(dockerFilePath)
+	if err != nil {
+		fmt.Printf("hover: Failed to create Dockerfile %s: %v\n", dockerFilePath, err)
+		os.Exit(1)
+	}
+	dockerFileContent := []string{
+		"FROM dockercore/golang-cross",
+		"RUN apt-get install libgl1-mesa-dev xorg-dev -y",
+		"WORKDIR /app",
+		"CMD " + strings.Join(buildCommand(targetOS, vmArguments, "build/outputs/"+targetOS+"/"+outputBinaryName(projectName, targetOS)), " "),
+	}
+	for _, env := range buildEnv(targetOS, "/engine") {
+		dockerFileContent = append(dockerFileContent, "ENV "+env)
+	}
+
+	for _, line := range dockerFileContent {
+		if _, err := dockerFile.WriteString(line + "\n"); err != nil {
+			fmt.Printf("hover: Could not write Dockerfile: %v\n", err)
+			os.Exit(1)
+		}
+	}
+	err = dockerFile.Close()
+	if err != nil {
+		fmt.Printf("hover: Could not close Dockerfile: %v\n", err)
+		os.Exit(1)
+	}
+	dockerBuildCmd := exec.Command(dockerBin, "build", "-t", "hover-build-cc", ".")
+	dockerBuildCmd.Stderr = os.Stderr
+	dockerBuildCmd.Stdout = os.Stdout
+	dockerBuildCmd.Dir = tmpDir
+	err = dockerBuildCmd.Run()
+	if err != nil {
+		fmt.Printf("hover: Docker build failed: %v\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Println("hover: Cross-Compiling 'go-flutter' and plugins using docker")
+
+	outputPath, err := filepath.Abs(filepath.Join(buildPath, "build", "outputs"))
+	dockerRunCmd := exec.Command(dockerBin, "run", "-e", "USERID=$UID", "-v", goPath+":/go", "-v", pkgRootDir+":/app", "-v", engineCachePath+":/engine", "-v", outputPath+":/app/build/outputs", "-v", filepath.Join(userCacheDir, "go-build")+":/cache", "hover-build-cc")
```

Comment:
only sharing the `pkgRootDir` (equal to `go`) causes problem on the [`plugin_tutorial`](https://github.com/go-flutter-desktop/examples/tree/master/plugin_tutorial)
```log
go: parsing /go-plugin-example/complex/go.mod: open /go-plugin-example/complex/go.mod: no such file or directory
go: parsing /go-plugin-example/battery/go.mod: open /go-plugin-example/battery/go.mod: no such file or directory
go: error loading module requirements
```

that because we are using `replace` in our `go.mod` ([code](https://github.com/go-flutter-desktop/examples/blob/28917b4976fe90debcc007d9041b21a940d09e15/plugin_tutorial/go/go.mod#L12-L14)) and point the plugin to the up directory.

I don't really know how to deal with this issue, we cannot mount the entire filesystem into the container (and even if we do, the replace path WILL differ in the container).

Reply:
By letting the user customize the Dockerfile, we could add other directories inside the container.
And let the user add external deps for plugin.

Reply:
The Dockerfile file can be saved into go/cross-compiling/windows/Dockerfile for example.
This way it can also be added to git

Reply:
It doesn't even need to be placed in a extra windows folder, because the Docker image is for general cross-compiling.

Reply:
Sorry I was wrong. For every OS a directory has to be created.

Reply:
By editing the docker file, we could mount the plugin source code into the container. OK got it!

Reply:
No no :) That's unnecessary. I justed mounted the whole app directory and changed the paths and it worked. So no custom volumes etc.

Reply:
Oh ok!